conans/model/profile.py

```diff
@@ -89,15 +90,28 @@ def dumps(self):
 
         return "\n".join(result).replace("\n\n", "\n")
 
-    def update(self, other):
+    def update_profile(self, other):
         self.update_settings(other.settings)
         self.update_package_settings(other.package_settings)
         # this is the opposite
         other.env_values.update(self.env_values)
         self.env_values = other.env_values
         self.options.update(other.options)
+        # It is possible that build_requires are repeated, or same package but different versions
         for pattern, req_list in other.build_requires.items():
-            self.build_requires.setdefault(pattern, []).extend(req_list)
+            existing_build_requires = self.build_requires.get(pattern)
+            existing = OrderedDict()
+            if existing_build_requires is not None:
+                for br in existing_build_requires:
+                    r = (ConanFileReference.loads(br)
+                         if not isinstance(br, ConanFileReference) else br)
```

Comment:
```suggestion
                    r = ConanFileReference.loads(br)
                         if not isinstance(br, ConanFileReference) else br
```

better if it doesn't look like a generator.

Reply:
It is a line break, otherwise it needs a backslash

Reply:
Added to the suggestion